Doc/glossary.rst

```diff
@@ -40,16 +40,18 @@ Glossary
       ABCs with the :mod:`abc` module.
 
    annotation
-      A metadata value associated with a global variable, a class attribute or a
-      function or method parameter or return value, that stores a
-      :term:`type hint`.
+      A label with no defined semantics associated with a variable, a class
```

Comment:
I think "no defined semantics" is too strong -- IMO it should say that by convention annotations are usually used for type hints (i.e. as type annotations).

Also I'm not too excited about the word "label" here but I agree it's better than "metadata value".

Reply:
What about "syntactic construct"? Would that be too wordy?